src/cogent3/maths/distance_transform.py

```diff
@@ -1253,7 +1267,41 @@ def binary_dist_hamming(datamtx, strict=True):
     return dists
 
 
-def binary_dist_jaccard(datamtx, strict=True):
+def fast_jaccard(x: iter, y: iter):
+    xset = set(x)
+    yset = set(y)
+    return 1 - len(xset & yset) / len(xset | yset)
+
+
+@singledispatch
+def jaccard(x, y):
+    """Jaccard distance between two sets is a measure of similarity between two sets
+    (0) identical, (1) dissimilar"""
+    raise NotImplementedError(f"jaccard distance not implemented for {type(x)}")
```

Comment:
Change the error message to
```python
return NotImplementedError(f"jaccard distance handles sets or numpy arrays, not {type(x)}")
```

Reply:
Yes, that's much more informative.  Will make the change.